RELEASE.md

```diff
@@ -2,22 +2,52 @@
 
-* To fix issues on existing release branches, fix commits must first be cherry-picked to the release branch.
-* A summary tag should be created, and signed explaining the fixes.
-* If the fix has security implications, it must be noted in the commit notes, such that the release alert provided by github to watchers can express this.
-* A post should be made on the website explaining the need for the bugfix release.
+## Release Flow
+
+### Major / Minor Releases
+
+#### Tagging an RC
+
+- [ ] A release is initiated by creating a protected release branch 'release-va.b' marking the major/minor release.
+- [ ] In preparation for a release, the maintainer should look at the commit log to create a change log summarizing changes, with explicit notes of:
+  - [ ] wire protocol changes that will impact group or node-client interactions.
+  - [ ] protocol changes that will impact setup or resharing.
+  - [ ] feature / behavior deviations.
+- [ ] Once this branch is created, protected, and pushed, a signed tag should be created to indicate the first release candidate 'va.b.0-rc1'.
+- [ ] This tag will be turned by the `go-releaser` CI action into a release candidate on github, and will notify anyone who is watching releases of the new action.
+
+#### Automated Testing
+
+- [ ] This release should go through:
+  - [ ] Unit testing
+  - [ ] Regression Testing
+  - [ ] Testground Testing
+  - [ ] Simulated Workflow (demo) Testing
+
+#### Testing in LoE Testnet
+
+- [ ] Reach out to the LoE partners to deploy the RC to Testnet. Ensuring that while the migration happens, there are no regressions between versions.
+- [ ] A minimum 1 week 'soak time' should follow a release candidate release before the first non-release-candidate `va.b.c` tag is pushed to create a full release on the new branch.
+
+#### Releasing and deploying to LoE Mainnet
+
+- [ ] Reach out to LoE partners to deploy the latest release
+- [ ] Releases should be communicated on the drand website through a post
+
+### Patch Releases (aka small bugfixes)
+
+- [ ] To fix issues on existing release branches, fix commits must first be cherry-picked to the release branch.
+- [ ] A summary tag should be created, and signed explaining the fixes.
+- [ ] A post should be made on the website explaining the need for the bugfix release.
+
+### Releases with Security implications
+
+If the fix has security implications, a smooth transition plan should be designed and communicated to LoE partners, so that we can ensure that the network is upgraded before sharing publicly of a known vulnerability on a service that users are depending on. We will handle these situations with extreme care and might be more quiet about them due to their nature. To learn how to report a vulnerability, please consult [SECURITY](./SECURITY.md)
```

Comment:
If I were contemplating a non-LoE deployment, this would make me nervous. We should have a plan for handling security implications with all known users, not just one.

(I understand that it's written above that this is the only major deployment so far -- but there is no reason to make the whole document so specific)

Reply:
agree. this also is not fully in sync with SECURITY.md, which outlines our security reporting / notification process.